src/contract_info.rs

```diff
@@ -252,6 +252,87 @@ pub fn migrate_contract_info(
     match (&msg.bid_fee_account, &msg.bid_fee_rate) {
         (Some(account), Some(rate)) => {
             contract_info.bid_fee_info = match (account.as_str(), rate.as_str()) {
+                ("", "") => None,
+                (_, _) => {
+                    Decimal::from_str(rate).map_err(|_| ContractError::InvalidFields {
+                        fields: vec![String::from("bid_fee_rate")],
+                    })?;
+
+                    Some(FeeInfo {
+                        account: api.addr_validate(account)?,
+                        rate: rate.to_string(),
+                    })
+                }
+            }
+        }
+        (_, _) => (),
+    };
+
+    match &msg.ask_required_attributes {
+        None => {}
+        Some(ask_required_attributes) => {
+            contract_info.ask_required_attributes = ask_required_attributes.clone();
+        }
+    }
+
+    match &msg.bid_required_attributes {
+        None => {}
+        Some(bid_required_attributes) => {
+            contract_info.bid_required_attributes = bid_required_attributes.clone();
+        }
+    }
+
+    set_contract_info(store, &contract_info)?;
+
+    get_contract_info(store)
+}
+
+pub fn modify_contract_info(
```

Comment:
this is essentially a copy of migrate_contract_info right?

wondering for the repeated pieces if we should move them to another fun 🤷